packages/io/src/io.ts

```diff
@@ -192,69 +188,89 @@ export async function which(tool: string, check?: boolean): Promise<string> {
         )
       }
     }
+
+    return result
   }
 
   try {
-    // build the list of extensions to try
-    const extensions: string[] = []
-    if (ioUtil.IS_WINDOWS && process.env.PATHEXT) {
-      for (const extension of process.env.PATHEXT.split(path.delimiter)) {
-        if (extension) {
-          extensions.push(extension)
-        }
-      }
+    const matches: string[] = await findInPath(tool)
+
+    if (matches && matches.length > 0) {
+      return matches[0]
     }
 
-    // if it's rooted, return it if exists. otherwise return empty.
-    if (ioUtil.isRooted(tool)) {
-      const filePath: string = await ioUtil.tryGetExecutablePath(
-        tool,
-        extensions
-      )
+    return ''
+  } catch (err) {
+    throw new Error(`which failed with message ${err.message}`)
+  }
+}
 
-      if (filePath) {
-        return filePath
-      }
+/**
```

Comment:
Most of the `which` method is now in `findInPath`.  Most of the changes are just indentation...use the "hide whitespace changes" option for easier viewing.